test/modules/paymentProcessor/bridging/abstracts/PP_Connext_Crosschain_v1_Test.t.sol

```diff
@@ -0,0 +1,457 @@
+        paymentProcessor = CrosschainBase_v1_Exposed(Clones.clone(impl));
+
+        //Setup the module to test
+        _setUpOrchestrator(paymentProcessor);
+
+        //General setup for other contracts in the workflow
+        _authorizer.setIsAuthorized(address(this), true);
+
+        //Initiate the PP with the medata and config data
+        paymentProcessor.init(_orchestrator, _METADATA, abi.encode(1));
+
+        //Setup other modules needed in the unit tests.
+        //In this case a payment client is needed to test the PP_Template_v1.
+        impl = address(new ERC20PaymentClientBaseV1Mock());
+        paymentClient = ERC20PaymentClientBaseV1Mock(Clones.clone(impl));
+        //Adding the payment client is done through a timelock mechanism
+        _orchestrator.initiateAddModuleWithTimelock(address(paymentClient));
+        vm.warp(block.timestamp + _orchestrator.MODULE_UPDATE_TIMELOCK());
+        _orchestrator.executeAddModule(address(paymentClient));
+
+        //Init payment client
+        paymentClient.init(_orchestrator, _METADATA, bytes(""));
+        paymentClient.setIsAuthorized(address(paymentProcessor), true);
+        paymentClient.setToken(token);
+        impl = address(new PP_Connext_Crosschain_v1());
+        crossChainManager = PP_Connext_Crosschain_v1(Clones.clone(impl));
+        // Setup mock addresses
+        mockConnextBridge = address(0x123456);
+        mockEverClearSpoke = address(everclearPaymentMock); // Using the existing mock
+        mockWeth = address(0x789012); // Or deploy a mock WETH contract if needed
+
+        //init the processor
+        // Initialize with proper config data
+        bytes memory configData = abi.encode(
+            address(mockEverClearSpoke), // address of your EverClear spoke contract
+            address(mockWeth) // address of your WETH contract
+        );
+
+        crossChainManager.init(_orchestrator, _METADATA, configData);
+        paymentClient.setIsAuthorized(address(crossChainManager), true);
+
+        // Setup token approvals and initial balances
+        token.mint(address(this), 1000 ether);
+        token.approve(address(crossChainManager), type(uint).max);
+
+        // Add these lines to ensure proper token flow
+        token.mint(address(crossChainManager), 1000 ether); // Mint tokens to processor
+        vm.prank(address(crossChainManager));
+        token.approve(address(crossChainManager), type(uint).max); // Processor approves bridge logic
+    }
+
+    function testInit() public override(ModuleTest) {
+        assertEq(
+            address(paymentProcessor.orchestrator()), address(_orchestrator)
+        );
+    }
+
+    function testReinitFails() public override(ModuleTest) {
+        vm.expectRevert(OZErrors.Initializable__InvalidInitialization);
+        paymentProcessor.init(_orchestrator, _METADATA, abi.encode(1));
+    }
+
+    function test_ProcessPayments_singlePayment() public {
```

Comment:
:see_no_evil: I couldnt resist

Reply:
HAHA this is great, good point out, didn't realize this was part of the style guide but think I have to revisit the code of conduct and do another review of this PR as there were a few points where it could have helped. Please review the gherkin we added and if you have any comments let us know marking as resolved for now though!